birthday/birthday.py

```diff
@@ -0,0 +1,259 @@
+import logging
+import hashlib
+import asyncio
+import contextlib
+import datetime
+import discord
+import itertools
+
+from redbot.core import commands, Config, checks
+from redbot.core.bot import Red
+from redbot.core.config import Group
+from redbot.core.i18n import Translator, cog_i18n
+from redbot.core.commands import Context, Cog
+
+T_ = Translator("Birthdays", __file__)
+
+def _(s):
+    def func(*args, **kwargs):
+        real_args = list(args)
+        real_args.pop(0)
+        return T_(s).format(*real_args, **kwargs)
+    return func
+
+@cog_i18n(T_)
+class Birthdays(Cog):
+    """Announces people's birthdays and gives them a birthday role for the whole day"""
+    __author__ = "PancakeSparkle#8243"
+
+    # Just some constants
+    DATE_GROUP = "DATE"
+    GUILD_DATE_GROUP = "GUILD_DATE"
+
+    # More constants
+    BDAY_LIST_TITLE = _("Birthday List")
+
+    # Even more constants
+    BDAY_WITH_YEAR = _("<@!{}> is now **{} years old**. <:aureliahappy:548738609763713035>")
+    BDAY_WITHOUT_YEAR = _("Everypony say Happy Hirthday to <@!{}>! <:aureliahappy:548738609763713035>")
+    ROLE_SET = _("<:aureliaagree:616091883013144586> The birthday role on **{g}** has been set to: **{r}**.")
+    BDAY_INVALID = _(":x: The birthday date you entered is invalid. It must be `MM-DD`.")
+    BDAY_SET = _("<:aureliaagree:616091883013144586> Your birthday has been set to: **{}**.")
+    CHANNEL_SET = _("<:aureliaagree:616091883013144586> "
+                    "The channel for announcing birthdays on **{g}** has been set to: **{c}**.")
+    BDAY_REMOVED = _(":put_litter_in_its_place: Your birthday has been removed.")
+
+
+    def __init__(self, bot):
+        super().__init__()
+        self.bot = bot
+        self.logger = logging.getLogger("aurelia.cogs.birthdays")
+        unique_id = int(hashlib.sha512((self.__author__ + "@" + self.__class__.__name__).encode()).hexdigest(), 16)
+        self.config = Config.get_conf(self, identifier=unique_id)
+        self.config.init_custom(self.DATE_GROUP, 1)
+        self.config.init_custom(self.GUILD_DATE_GROUP, 2)
+        self.config.register_guild(channel=None, role=None, yesterdays=[])
+        self.bday_loop = asyncio.ensure_future(self.initialise())
+        asyncio.ensure_future(self.check_breaking_change())
+
+    # Events
+    async def initialise(self):
+        await self.bot.wait_until_ready()
+        with contextlib.suppress(RuntimeError):
+            while self == self.bot.get_cog(self.__class__.__name__):
+                now = datetime.datetime.utcnow()
```

Comment:
need to add in the clean/do birthdays _before_ the sleep for a day, since if the bot restarts, or someone adds their birthday as the day today, then it won't trigger. I would also consider running do_today_bdays when someone set's their birthday as well.